### PR TITLE
Add more Halstead metrics

### PR DESCRIPTION
**Pull request by tuncbkose**
Hi!

This is related to issue #1883. I just copied what was happening with the previous metrics, let me know if I misunderstood what had to be done (This is my first contribution and I am not very clear on what the rest of the code does).

### CODE REVIEW

**Comment by tuncbkose** [comment]
Noticed the failing tests. Will work on it and if appropriate add more tests.

**Review by marco-c — APPROVED** [review]
I won't merge it yet as it requires a DB version bump (which I don't want to do at the moment), but it LGTM, thanks!

**Pull request merged**
